index.tsx

```diff
@@ -116,16 +117,19 @@ class AnimateOnChange extends Component<Props, State> implements AnimateOnChange
   }
 
   render () {
-    let className = this.props.baseClassName
+    const { clearAnimationClass } = this.state;
+    const { baseClassName, animate, animationClassName, customTag, children, otherProps } = this.props;
```

Comment:
Should probably use the [spread operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#object-spread-and-rest) for `otherProps`, such that one does not have to pass props under the otherProps namespace.

E.g.
```jsx
<AnimateOnChange otherProps={{ id: 1 }}>hello</AnimateOnChange>
```
vs
```jsx
<AnimateOnChange id=1>hello</AnimateOnChange>
```

Reply:
Yes, it would be much better. And I actually wanted to do it initially, but I could not find a way to correctly write the types for the rests operator.
```ts
export interface Props {
  children: any,
  animate: boolean,
  baseClassName: string,
  animationClassName: string,
  customTag?: string,
  onAnimationEnd?: () => void,
  // Am I here to describe all the possible custom props?
}
```
I am not very familiar with strict typing. Maybe you know how to do it right?

```js
<AnimateOnChange
  id=1
  onClick={handleClick}
  onMouseOver={this.handleMouseOver} 
  onMouseOut={this.handleMouseOut}
>
  hello
</AnimateOnChange>

/////////////////////////////////////////

render () {
    const { clearAnimationClass } = this.state;
    const { baseClassName, animate, animationClassName, customTag, children, ...otherProps } = this.props;

    let className = baseClassName

    if (animate && !clearAnimationClass) {
      className += ` ${animationClassName}`
    }

    let Tag = customTag || 'span';

    return <Tag ref={this.setElementRef} className={className} {...otherProps}>
      {children}
    </Tag>
  }
```

Reply:
I believe you can use an intersection type, maybe `Props & any`?

```
class AnimateOnChange extends Component<Props & any, State> implements AnimateOnChange {
```

Reply:
Yes it works thanks for the advice)